### PR TITLE
Make Teleport version requirements more consistent

### LINKED ISSUE #40886: Make version requirements for minor Teleport versions in docs pages more consistent
## Applies To

Docs pages for functionality available in a minor versions of Teleport.

## Details

If we introduce functionality in a minor version of Teleport, we need to add a note to the docs to specify the minor version that introduces the functionality, since users cannot infer the minor version from the version of the docs site. We use a couple of methods to do this:

- `Admonition` components
- Items in a Prerequisites section
- Potentially others

Let's make this consistent.

## How will we know this is resolved?

Mentions of required minor versions in docs pages use the same format.

### PR DESCRIPTION
**Pull request by ptgott**
Closes #40886

When a how-to guide mentions the need for a specific Teleport version, ensure that the Prerequisites section of the guide includes the version requirement. This keeps all prerequisites for the guide in one place and avoids surprising the user with a version requirement at an unexpected location within the guide.

Also remove mentions of Teleport versions before v13, which are unsupported in the docs. Don't touch reference guides, which we do not expect to include a Prerequisites section.

### CODE REVIEW

File: docs/pages/access-controls/device-trust/guide.mdx [hunk 1]
```diff
@@ -45,54 +45,53 @@ protected with Teleport.
   root@(=clusterDefaults.nodeIP=):~#
   ```
 
+<Details type="warning" title="Teleport v13.3.5 and Below">
```
Comment (zmb3) [inline docs/pages/access-controls/device-trust/guide.mdx:48]: Do we need to mention v13 anymore?

File: docs/pages/access-controls/guides/headless.mdx [hunk 2]
```diff
@@ -25,17 +25,17 @@ For example:
 
 ## Prerequisites
 
-- A v12.2+ Teleport cluster with WebAuthn configured.
+- A Teleport cluster with WebAuthn configured.
   See the [Second Factor: WebAuthn](./webauthn.mdx) guide.
 - WebAuthn hardware device, such as YubiKey.
-- Machines for Headless WebAuthn activities have [Linux](../../installation.mdx#linux), [macOS](../../installation.mdx#macos) or [Windows](../../installation.mdx#windows-tsh-and-tctl-clients-only) `tsh` binary v12.2+ installed.
+- Machines for Headless WebAuthn activities have [Linux](../../installation.mdx#linux), [macOS](../../installation.mdx#macos) or [Windows](../../installation.mdx#windows-tsh-client-only) `tsh` binary installed.
```
Comment (zmb3) [inline docs/pages/access-controls/guides/headless.mdx:31]: This looks like a regression (broken link).

Comment (mmcallister) [reply docs/pages/access-controls/guides/headless.mdx:31]: It looks like `## Windows (tsh and tctl clients only)` is the correct header

File: docs/pages/access-controls/device-trust/guide.mdx (hunk 1 above)
Comment (ptgott) [reply docs/pages/access-controls/device-trust/guide.mdx:48]: Our convention has been to remove version warnings for versions below the latest EOL version: https://github.com/gravitational/teleport/blob/master/.github/ISSUE_TEMPLATE/test-plan-docs.md#is-the-docs-site-up-to-date-with-the-new-release

**Review by zmb3 — APPROVED** [review]

File: docs/pages/includes/tls-multiplexing-warnings.mdx [hunk 3]
```diff
@@ -1,52 +0,0 @@
...
-
-```yaml
-annotations:
-  ingress:
-    nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
-```
-
-Deploying Teleport behind Cloudflare, whether using its proxy ("orange-clouding") or tunnels
-(`cloudflared`) should work with Teleport version 15.1 or higher. See the
-[TLS Routing FAQ](../reference/architecture/tls-routing.mdx) for more details.
```
Comment (nklaassen) [inline docs/pages/includes/tls-multiplexing-warnings.mdx:21]: this section seems relevant for 14.0+, should we keep it and delete the rest?

**Comment by ptgott** [comment]
@nklaassen Following up to see if I have resolved your feedback. Thanks!

**Review by stevenGravy — APPROVED** [review]

File: docs/pages/admin-guides/access-controls/device-trust/guide.mdx [hunk 4]
```diff
@@ -45,54 +45,53 @@ protected with Teleport.
...
+
+  For simplicity, the example below updates the preset `access` role but you can update
+  any existing access granting role which the user is assigned with to enforce Device Trust.
+
+  First, fetch a role so you can update it locally:
+  ```code
+  $ tctl edit role/access
+  ```
+
+  Edit the role with device trust mode:
```
Comment (mmcallister) [inline docs/pages/admin-guides/access-controls/device-trust/guide.mdx:61]:
```suggestion
  Edit the role with Device Trust mode:
```

**Pull request merged**